### PR TITLE
CLI checks for AWS credentials

### LINKED ISSUE #713: Run AWS credentials check automatically via CLI
## This is:
- Feature request

## Expected Behavior
In order to make DX better, let's automatically execute the following check for our users, and throw a clear error message if the credentials are incorrect / missing.

image

### PR DESCRIPTION
**Pull request by TommyJackson85**
## Related Issue

Closes #713

## Your solution

Checking for aws sts credentials from a new plugin.

## How Has This Been Tested?
Regularly ran ```npx webiny deploy api --env=local``` from sample-project folder
Console logged succesfull passes originally and error messages originally.

Will continue to look through aws documentation to double check my methods and what I need to do next.

### CODE REVIEW

File: packages/cli-plugin-deploy-components/aws/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//const execa = require('execa');
+//const AWS_CREDENTIALS = execa("aws sts get-caller-identity"); \
+const STS = require("aws-sdk/clients/sts");
+
+module.exports = {
+    type: "hook-before-deploy",
+    name: "cli-command-aws",
```
Comment (Pavel910) [inline packages/cli-plugin-deploy-components/aws/index.js:7]: Change the name of the plugin to `hook-before-deploy-aws-credentials`

File: packages/cli-plugin-deploy-components/aws/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                    console.log(`Please to refer to the following link for setting up credentials:
+                    Please to refer to the following link for setting up aws credentials:
+                    https://docs.webiny.com/docs/guides/aws-credentials/
+                    `);
+                } else {
+                    console.log("Your AWS data is set!");
+                    console.log(result);
+                }
+            });
+    }
```
Comment (Pavel910) [inline packages/cli-plugin-deploy-components/aws/index.js:26]: This part can be written a lot simpler and cleaner:

```js
try {
    await sts.getCallerIdentity({}).promise();
} catch(err) {
    console.log(`Looks like your AWS credentials are not configured correctly!`);
    // .... link to docs ....
}
```

Comment (TommyJackson85) [reply packages/cli-plugin-deploy-components/aws/index.js:26]: @Pavel910 sorry! I misinterpreted one of your posts from the issue post. I will bring back try and catch so.

Comment (TommyJackson85) [reply packages/cli-plugin-deploy-components/aws/index.js]: @Pavel910 Do you want a message to appear if aws credentials pass?

**Comment by TommyJackson85** [comment]
@Pavel910 Do you want a message to appear if aws credentials pass?

**Comment by Pavel910** [comment]
@TommyJackson85 no need, we don't want to spam users if there's no actual use for that message. If credentials are fine - just move on with the process. If not - that's where we want user's attention.

File: packages/cli-plugin-deploy-components/aws/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    name: "hook-before-deploy-aws-credentials",
+    async hook() {
+        // check AWS credentials
+        const sts = new STS();
+        try {
+            await sts.getCallerIdentity({}).promise();
+        } catch (err) {
+            console.log("Looks like your AWS credentials are not configured correctly!");
+            console.log(
+                "Please to refer to the following link for setting them up: https://docs.webiny.com/docs/guides/aws-credentials/"
```
Comment (Pavel910) [inline packages/cli-plugin-deploy-components/aws/index.js:14]: This sentence looks off to me: `Please to refer to the following...` ?

**Comment by TommyJackson85** [comment]
@Pavel910 any suggestions on the kind of message we should display? I updated it, as I think you were right, it was too complex. It was a little bit long too. Let me know what you think of my changes.

**Comment by Pavel910** [comment]
@TommyJackson85 it wasn't too long, just had an extra `to` which made it totally weird.

I think we can do something like: 
```
To learn how to configure your AWS credentials, visit https://docs.webiny.com/docs/guides/aws-credentials.
```

**Comment by TommyJackson85** [comment]
@Pavel910 So I updated the console message again, and I removed the comment from the hook plugin and I removed the aws.js file as it ended up not being used and was empty.
Was ready to suggest a merge but now I have build package -> commodo error for some reason :

Screenshot 2020-07-10 at 19 28 51

I dont see why I would get this as it didn't exist before in previous checks. Any suggestions?

**Comment by doitadrian** [comment]
Try removing this: `typings/shared/index.d.ts:30`.

**Comment by TommyJackson85** [comment]
> Try removing this: `typings/shared/index.d.ts:30`, we added this a couple of days ago. There was a TS error in a 3rd party `@types` package, which was preventing us from doing a release.
> 
> Maybe they've fixed it already.
> 
> UPDATE: it looks like the issue got closed / resolved ([DefinitelyTyped/DefinitelyTyped#45927](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927)).

Well for further context, here is the file you suggested, from my repository:

Screenshot 2020-07-10 at 21 08 43

I did also notice there were some file setup changes , like the removal of sample-project folder, bringing apps folder to the root folder. Could this be related to it?

**Comment by TommyJackson85** [comment]
I pulled the master branch again just to keep it updated. But I had to fix an error in a handler.js file to do with no curly braces, in order to commit my changes. Lets hope this passes!

**Comment by TommyJackson85** [comment]
same error again

**Comment by TommyJackson85** [comment]
> Try removing this: `typings/shared/index.d.ts:30`, we added this a couple of days ago. There was a TS error in a 3rd party `@types` package, which was preventing us from doing a release.
> 
> Maybe they've fixed it already.
> 
> UPDATE: it looks like the issue got closed / resolved ([DefinitelyTyped/DefinitelyTyped#45927](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927)).

@doitadrian Is the file ok here? This is the most updated version:

Screenshot 2020-07-11 at 01 18 36

**Comment by doitadrian** [comment]
Sorry, I'm kinda confused. Did you try removing the `typings/shared/index.d.ts:30`?

Thanks!

**Comment by TommyJackson85** [comment]
> Sorry, I'm kinda confused. Did you try removing the `typings/shared/index.d.ts:30`?
> 
> Thanks!
> 
> UPDATE: from what I can see, this wasn't removed. Please remove it and try again.

You said the issue was resolved and I pulled the most recent master branch. No worries though, I removed the interface Module at line 30, as you were referring to that. If it doesn't work, I will try and remove the namespace NodeJS object.

**Comment by Pavel910** [comment]
@TommyJackson85 Adrian was referring to this issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927.
It's the reason why that TS error is happening in the first place.

Since you pulled the changes from Webiny `master` today and installed the dependencies *today*, it's possible that the underlying issue linked above was resolved and now our hotfix is causing the clash with the original code that had to be fixed.

I'll take a look at the code and merge it later, thanks!

**Comment by TommyJackson85** [comment]
@Pavel910 no problem. It would be good to double check my file changes so they don't cause errors but they're passing all checks now.

File: typings/shared/index.d.ts [hunk 4]
```diff
@@ -27,7 +27,4 @@ declare module "*.svg" {
 
 // Hot fix - check https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927 for a fix.
```
Comment (doitadrian) [inline typings/shared/index.d.ts:28]: Please delete this while section, including the comment and the `declare namespace ...` statement.

**Comment by TommyJackson85** [comment]
> Minor change, and if you ask me, this looks good.

@doitadrian Will get around to this later. I apologise for the delay!

**Comment by TommyJackson85** [comment]
@doitadrian deleted what you suggested, just double check it to make sure.

**Pull request merged**